crates/editor/src/element.rs

```diff
@@ -1482,6 +1482,64 @@ impl EditorElement {
         axis_pair(horizontal_scrollbar, vertical_scrollbar)
     }
 
+    fn layout_minimap(
+        &self,
+        window: &mut Window,
+        cx: &mut App,
+        snapshot: &EditorSnapshot,
+        bounds: Bounds<Pixels>,
+        scrollbars_layout: AxisPair<Option<ScrollbarLayout>>,
+        line_height: Pixels,
+    ) -> Option<AnyElement> {
+        match snapshot.mode {
+            EditorMode::Full => {
+                let mut editor = self
+                    .editor
+                    .update(cx, |editor, cx| editor.clone(window, cx));
+
+                editor.mode = EditorMode::Minimap;
+                editor.set_text_style_refinement(TextStyleRefinement {
+                    font_size: Some(px(2.).into()),
+                    font_weight: Some(gpui::FontWeight(900.)),
+                    ..Default::default()
+                });
+
+                let editor_entity = cx.new(|_| editor);
+
+                let mut minimap_elem = editor_entity.update(cx, |editor, cx| {
+                    editor.render(window, cx).into_any_element()
+                });
+
+                let scrollbar_y_width = match scrollbars_layout.vertical {
+                    Some(scrollbar) => {
+                        if scrollbar.visible {
+                            scrollbar.hitbox.bounds.size.width
+                        } else {
+                            px(0.)
+                        }
+                    }
+                    None => px(0.),
+                };
+                let mut bottom_right = bounds.bottom_right();
+                bottom_right.x -= scrollbar_y_width;
+                let header_height = line_height * FILE_HEADER_HEIGHT as f32;
+                let minimap_bounds = Bounds::from_corners(
+                    point(
+                        bounds.size.width - px(100.) - scrollbar_y_width,
+                        bounds.origin.y - header_height + px(2.),
```

Comment:
@SomeoneToIgnore I had offset the minimap y origin to account for the height of the buffer header, I'm not sure if there's a cleaner way to do this. Could you take a look?

Reply:
Does not look too bad, at least for the first iteraion, where we concentrate on the basic functionality.

Reply:
I figured out a better and more reliable way to get the positioning by basing it directly on the vertical scrollbar bounds. If the scrollbar is None, I don't render the minimap, since right now I'm only targeting minimap for the Full editor and AFAICT the scrollbar bounds are still calculated for Full editor even if the scrollbar is set to not show.